### PR TITLE
common/features: only support a single feature bitset.

### PR DESCRIPTION
**Pull request by rustyrussell**
This is mainly an internal-only change, especially since we don't
offer any globalfeatures.

However, LND (as of next release) will offer global features, and also
expect option_static_remotekey to be a *global* feature.  So we send
our (merged) feature bitset as both global and local in init, and fold
those bitsets together when we get an init msg.

This works with every proposal:

https://github.com/lightningnetwork/lightning-rfc/pull/680
https://github.com/lightningnetwork/lightning-rfc/pull/666

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 Note: You should always set `allow-deprecated-apis=false` to test for
 changes.
 
+- JSON API: `listpeers` and `listnodes` fields `localfeatures` and `globalfeatures` (now just `features`).
```
Comment (darosior) [inline CHANGELOG.md]: Should we also change the plugin hooks/notifications API then ?

Comment (darosior) [reply CHANGELOG.md]: (`peer_connected` for example)

File: connectd/connectd.h [hunk 2]
```diff
@@ -19,7 +19,6 @@ struct io_plan *peer_connected(struct io_conn *conn,
 			       const struct node_id *id,
 			       const struct wireaddr_internal *addr,
 			       const struct crypto_state *cs,
-			       const u8 *globalfeatures TAKES,
 			       const u8 *localfeatures TAKES);
```
Comment (niftynei) [inline connectd/connectd.h]: localfeatures -> features ?

File: CHANGELOG.md (hunk 1 above)
Comment (niftynei) [reply CHANGELOG.md]: `peer_connected` was updated in 52207acd9fe6b0cb858668e081fbb92a7f77a024, the documentation is lagging however.

**Comment by niftynei** [comment]
went ahead and fixed up my two nits; updated commit is here https://github.com/niftynei/lightning/commit/ae6a3dbd8eefafc7c81de6564eaae56f7f950e1b

**Comment by rustyrussell** [comment]
OK, that was weird.  Rebased on master, pushed a fixup for review (squash on merge please!)

**Pull request merged**